### PR TITLE
[ODS-4874] WebAPI installer does not enable any features 

### PR DESCRIPTION
**Pull request by semalaiappan**
$parameters = @{
      DbConnectionInfo = @{
        Engine="SqlServer"
        Server="localhost"
        UseIntegratedSecurity=$true
    }
    InstallType = "Sandbox"
    WebApiFeatures = @{
            BearerTokenTimeoutMinutes="BearerTokenTimeoutMinutes"                                   
            ExcludedExtensionSources="GrandBend"
            Features= @(
                @{
                    Name= "changeQueries"
                    IsEnabled= $true
                    },
				@{
                    Name= "OpenApiMetadata"
                    IsEnabled= $true
                 },
				@{
                    Name= "composites"
                    IsEnabled= $true
                 },
				 @{
                    Name= "profiles"
                    IsEnabled= $true
                 },
				 @{
                    Name= "identityManagement"
                    IsEnabled= $true
                    },
				 @{
                    Name= "extensions"
                    IsEnabled= $true
                    },
				@{
                    Name= "ownershipBasedAuthorization"
                    IsEnabled= $true
                    },
				@{
                    Name= "uniqueIdValidation"
                    IsEnabled= $true
                 }
           )
     }      
}

### CODE REVIEW

File: Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1 [hunk 1]
```diff
@@ -390,12 +396,18 @@ function Invoke-TransformWebConfigAppSettings {
...
+                foreach ($feature in $Config.WebApiFeatures.Features) {
+                      foreach ($defaultfeature in $settings.ApiSettings.Features) {
+                         If ( $feature.Name -eq $defaultfeature.Name) {
+                             $defaultfeature.IsEnabled =$feature.IsEnabled
+                         }
+                      }
+                }
+            }
+           $settings.ApiSettings.BearerTokenTimeoutMinutes=$Config.WebApiFeatures.BearerTokenTimeoutMinutes
+           $settings.ApiSettings.ExcludedExtensions=$Config.WebApiFeatures.ExcludedExtensions
```
Comment (tbmeador-dlp) [inline Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1:409]:
```suggestion
            If ($Config.WebApiFeatures.Features -ne $Null -and $Config.WebApiFeatures.Features -ne '') {
                foreach ($feature in $Config.WebApiFeatures.Features) {
                    foreach ($defaultfeature in $settings.ApiSettings.Features) {
                        If ( $feature.Name -eq $defaultfeature.Name) {
                            $defaultfeature.IsEnabled =$feature.IsEnabled
                        }
                    }
                }
            }
           $settings.ApiSettings.BearerTokenTimeoutMinutes=$Config.WebApiFeatures.BearerTokenTimeoutMinutes
           $settings.ApiSettings.ExcludedExtensions=$Config.WebApiFeatures.ExcludedExtensions
```

File: Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1 [hunk 2]
```diff
@@ -390,12 +396,18 @@ function Invoke-TransformWebConfigAppSettings {
             $settings = Get-Content $settingsFile | ConvertFrom-Json | ConvertTo-Hashtable
             $settings.ApiSettings.Mode = $Config.InstallType
             $settings.ApiSettings.Engine = $Config.engine
-            $settings.ApiSettings.Features=$Config.WebApiFeatures.Features
-            $settings.ApiSettings.BearerTokenTimeoutMinutes=$Config.WebApiFeatures.BearerTokenTimeoutMinutes
-            $settings.ApiSettings.ExcludedExtensions=$Config.WebApiFeatures.ExcludedExtensions
-            $EmptyHashTable=@{}
-            $mergedSettings = Merge-Hashtables $settings, $EmptyHashTable
-            New-JsonFile $settingsFile $mergedSettings -Overwrite
+            If ( $Config.WebApiFeatures.Features -ne $Null -and $Config.WebApiFeatures.Features -ne '') {
```
Comment (tbmeador-dlp) [inline Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1:399]: do we need to check `Features -ne ''` here?

File: Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1 (hunk 1 above)
Comment (tbmeador-dlp) [reply Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1:409]: also can we update the example up above to match what the features array should look like:
https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/blob/e8f7c159647a92ca706a36a359adbdec092c4748/Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1#L134-L148

File: Scripts/NuGet/EdFi.Installer.SandboxAdmin/Install-EdFiOdsSandboxAdmin.psm1 [hunk 3]
```diff
@@ -75,6 +75,10 @@ function Install-EdFiOdsSandboxAdmin {
         [string]
         $PackageVersion,
 
+        # NuGet package source . default value is set for release package source for installer .
```
Comment (tbmeador-dlp) [inline Scripts/NuGet/EdFi.Installer.SandboxAdmin/Install-EdFiOdsSandboxAdmin.psm1:78]: can we remove the changes that belong to the other PR

**Review by tbmeador-dlp — CHANGES_REQUESTED** [review]

File: Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1 (hunk 2 above)
Comment (semalaiappan) [reply Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1]: No . I removed it . Thanks

File: Scripts/NuGet/EdFi.Installer.SandboxAdmin/Install-EdFiOdsSandboxAdmin.psm1 (hunk 3 above)
Comment (semalaiappan) [reply Scripts/NuGet/EdFi.Installer.SandboxAdmin/Install-EdFiOdsSandboxAdmin.psm1]: I removed it . Thanks

File: Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1 (hunk 1 above)
Comment (tbmeador-dlp) [reply Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1]: this docs still need to be updated here

File: Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1 [hunk 4]
```diff
@@ -390,12 +390,18 @@ function Invoke-TransformWebConfigAppSettings {
...
+            If ($Config.WebApiFeatures.Features -ne $Null) {
+                foreach ($feature in $Config.WebApiFeatures.Features) {
+                    foreach ($defaultfeature in $settings.ApiSettings.Features) {
+                        If ( $feature.Name -eq $defaultfeature.Name) {
+                            $defaultfeature.IsEnabled =$feature.IsEnabled
+                        }
+                    }
+                }
+            }
+           $settings.ApiSettings.BearerTokenTimeoutMinutes=$Config.WebApiFeatures.BearerTokenTimeoutMinutes
```
Comment (tbmeador-dlp) [inline Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1:402]: sorry I didn't catch this before but `BearerTokenTimeoutMinutes` should be up one level higher and not under `ApiSettings` for the api to be able to pick it up see:
https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/blob/1ff8abc70dde9c997ca8acf3c5a22436ebb18969/Application/EdFi.Ods.WebApi/appsettings.json#L15

**Review by tbmeador-dlp — CHANGES_REQUESTED** [review]

File: Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1 (hunk 4 above)
Comment (semalaiappan) [reply Scripts/NuGet/EdFi.Installer.WebApi/Install-EdFiOdsWebApi.psm1]: That's OK . I updated it . Thanks

**Comment by semalaiappan** [comment]
@tbmeador-dlp  All review comments are fixed .

**Review by tbmeador-dlp — APPROVED** [review]

**Pull request merged**